### ISSUE TITLE
[Desktop] Content Settings crash in Beta

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#10297]
```
[ 00 ] base::Value::Clone() const
[ 01 ] content_settings::(anonymous namespace)::CloneRule(content_settings::Rule const&, bool)
[ 02 ] content_settings::(anonymous namespace)::BraveShieldsRuleIterator::Next()
[ 03 ] HostContentSettingsMap::GetContentSettingValueAndPatterns(content_settings::RuleIterator*, GURL const&, GURL const&, ContentSettingsPattern*, ContentSettingsPattern*)
[ 04 ] HostContentSettingsMap::GetContentSettingValueAndPatterns(content_settings::ProviderInterface const*, GURL const&, GURL const&, ContentSettingsType, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, ContentSettingsPattern*, ContentSettingsPattern*)
[ 05 ] HostContentSettingsMap::GetWebsiteSettingInternal(GURL const&, GURL const&, ContentSettingsType, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, HostContentSettingsMap::ProviderType, content_settings::SettingInfo*) const
[ 06 ] HostContentSettingsMap::GetWebsiteSetting(GURL const&, GURL const&, ContentSettingsType, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, content_settings::SettingInfo*) const
[ 07 ] content_settings::CookieSettings::GetCookieSettingInternal(GURL const&, GURL const&, bool, content_settings::SettingSource*, ContentSetting*) const
[ 08 ] content_settings::CookieSettingsBase::IsCookieSessionOnly(GURL const&) const
[ 09 ] content::ServiceWorkerRegistry::OnStoragePolicyChanged()
[ 10 ] base::TaskAnnotator::RunTask(char const*, base::PendingTask*)
[ 11 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
[ 12 ] base::MessagePumpLibevent::Run(base::MessagePump::Delegate*)
[ 13 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta)
[ 14 ] base::RunLoop::Run()
[ 15 ] content::BrowserProcessSubThread::IOThreadRun(base::RunLoop*)
[ 16 ] base::Thread::ThreadMain()
[ 17 ] base::(anonymous namespace)::ThreadFunc(void*)
[ 18 ] 0x7f6353eca6db
[ 19 ] 0x7f634d94088f
```
https://brave.sp.backtrace.io/p/brave/debug?filters=(_deleted%3D0)&fingerprint=ac40b496a347e436c873dd79cf2235f76929fa37b21bfee18c12d0f85353a3fc&debug=(%223b545%22,0,0)

also 

https://brave.sp.backtrace.io/p/brave/debug?filters=(_deleted%3D0)&fingerprint=662cceffd2f85ff22ad4b6d7c25158b3bb7c7259455babc64726dfaf97ebafc8&debug=(%223b3ae%22,0,0)

first seen 28 May 2020

**Comment by iefremov** [comment]
@bsclifton @bridiver CC

**Comment by bridiver** [comment]
this crash has been around a long time, but my PR never got merged so rebasing and letting CI run https://github.com/brave/brave-core/pull/4653

**Comment by bsclifton** [comment]
@bridiver can we close this as https://github.com/brave/brave-core/pull/4653 was merged? Or did we need to verify on a new Nightly (can't hurt)

**Comment by bridiver** [comment]
we can close and then see if there are new crashes in backtrace

**Comment by iefremov** [comment]
@bridiver jfyi crashes from backtrace mentioned in the description are not that old - first seen end of May. Maybe there is something else

**Comment by bridiver** [comment]
@iefremov maybe it's a slightly different stack trace, but this crash has been around for a very long time

**Comment by kjozwiak** [comment]
@iefremov @bridiver let me know if this one should be closed as we merged https://github.com/brave/brave-core/pull/5867 into `1.11.x`.

**Comment by bridiver** [comment]
@kjozwiak I think we should close it and re-open if we see any more related crashes

**Comment by kjozwiak** [comment]
Quickly went through BackTrace and I don't see the following two appear after `1.12.31` which included this fix 👍 

* https://brave.sp.backtrace.io/p/brave/explore?time=month&aggregations=((timestamp%2Cbin%2C32)%2C(callstack%2Chead))&fingerprint=ac40b496a347e436c873dd79cf2235f76929fa37b21bfee18c12d0f85353a3fc

**Comment by iefremov** [comment]
@bridiver @kjozwiak I probably don't understand our versioning, but it appears to me that we still have this crash in versions 84.1.12.91, 84.1.12.42 etc

https://brave.sp.backtrace.io/p/brave/explore?filters=((callstack%2Ccontains%2Ccontent_settings)%2C(ver%2Ccontains%2C%221.12%22))&aggregations=((ver%2Cdistribution)%2C(callstack%2Chead))

**Comment by kjozwiak** [comment]
@iefremov going to close this off and create another issue. Once code is merged/released in a certain milestone, we usually create follow ups 👍 I'll take a look tomorrow and create a new issue and CC yourself and @bridiver.

**Comment by iefremov** [comment]
> @iefremov going to close this off and create another issue. Once code is merged/released in a certain milestone, we usually create follow ups 👍 I'll take a look tomorrow and create a new issue and CC yourself and @bridiver.

thanks @kjozwiak !
